test/unit/version_test.rb

```diff
@@ -190,6 +190,51 @@ class VersionTest < ActiveSupport::TestCase
       @version = build(:version)
     end
 
+    context "#rely_on_built_at?" do
+      setup do
+        created_at = Date.parse("2009-07-25")
+        built_at = created_at - 30.days
+        @version.update(created_at: created_at, built_at: built_at)
+      end
+
+      should "return true created_at date is equal 2009-07-25 and built_at is present and not higher than created_at" do
+        assert_predicate @version, :rely_on_built_at?
+      end
+
+      should "return false if built_at is not set" do
+        @version.update(built_at: nil)
+        refute_predicate @version, :rely_on_built_at?
+      end
+
+      should "return false if created_at is not 2009-07-25" do
+        @version.update(created_at: Date.parse("2020-01-01"))
+        refute_predicate @version, :rely_on_built_at?
+      end
+
+      should "return false if built_at is higher than 2009-07-25" do
+        @version.update(built_at: Date.parse("2020-01-01"))
+        refute_predicate @version, :rely_on_built_at?
+      end
+    end
+
+    context "#authored_at" do
+      setup do
+        @built_at = DateTime.parse("2010-01-01")
+        @created_at = DateTime.parse("2000-01-01")
+        @version.update(built_at: @built_at, created_at: @created_at)
+      end
+
+      should "return built_at as if #rely_on_built_at? returns true" do
+        @version.stubs(:rely_on_built_at?).returns(true)
```

Comment:
Definitely an option, I guess it's a question of how "unit" we want the test to be. Or, in other words, do we want this test to fail in case if `rely_on_built_at` definition changes. From purely unit-test point of view, the test should keep passing as long as the `authored_at` relies on `rely_on_built_at?`. But from an integration point of view, it would be better to rely on actual dates instead of stubs
I'll change it to rely on a date! Thanks